### ISSUE TITLE
NeoVim question and request, and an idea

### ISSUE THREAD

**Issue by surrain** [nicknisi/dotfiles#21]
Nick,

I've been enjoying your dotfiles and have a question, a humble request, and something like a question or a request (depending on how it's entangled).

I can't figure out what is responsible for appending end quotes and closing braces in NeoVim when I type their respective counterparts. If it's a setting in the config file, can you point it out to me?

Perhaps related, if it wouldn't be too much trouble, I'd love even a few words about the plugins. Most are uncommented, but a few are perfect, like these:

```
Plug 'tpope/vim-sleuth' " detect indent style (tabs vs. spaces)
Plug 'sickill/vim-pasta' " context-aware pasting
```

Thirdly, this might be a bad idea, but I'm wondering if you considered having your install script move existing dotfiles into a "displaced" folder instead of skipping them. Just a thought (unless you like the idea, in which case it's a humble request).

Regards,
Aaron

**Comment by nicknisi** [comment]
Thanks for the requests! I have gone through and added short comments next to all plugins.

The plugins that are responsible for appending quotes, braces, parens, etc is [Raimondi/delimitMate](https://github.com/Raimondi/delimitMate).

Finally, I really like the request to update the install script to backup existing config files and then to let the install script create the symlinks. I'll work on this! Please let us know if you have any other questions. Thanks!

**Comment by surrain** [comment]
Thank you!

**Comment by nicknisi** [comment]
@surrain I added `install/backup.sh`, which will copy all non-symlinked files that would be installed by this dotfiles setup into a `~/dotfiles-backup` directory. Hope this helps!

**Comment by surrain** [comment]
I’m psyched to try this stuff out. Thanks.

I hope you’ll trust me when I say I don’t solicit tech support from 
every rando on the internet, but I’m using a lightly moified version 
of your dotfiles so I figured this might be something you ran into. Have 
you ever had the problem where the delete key in neovim normal mode 
sends you a pane to the left? I tried setting iterm2 to send ^H on 
delete instead, but it deletes text normally in insert mode and skips 
out of the pane in normal mode either way, even after an iterm restart.

Thanks,
Aaron
